SecuritySystemUWP/SecuritySystemUWP/Storage/OneDrive.cs

```diff
@@ -173,297 +152,10 @@ public int GetNumberOfUploadedPictures()
-                        }
-
-                    }
-                }
-            }
-            catch (TaskCanceledException ex)
-            {
-                // Log telemetry event about this exception
-                var events = new Dictionary<string, string> { { "OneDrive", ex.Message } };
-                App.Controller.TelemetryClient.TrackEvent("CancelledFileUpload", events);
-
-                throw new Exception("SendFileAsync() - " + ex.Message);
-            }
-            catch (Exception ex)
-            {
-                // This failure will already be logged in telemetry in the enclosing UploadPictures function. We don't want this to be recorded twice.
-
-                throw new Exception("SendFileAsync() - Error: " + ex.Message);
-            }
-            finally
-            {
-                streamContent.Dispose();
-                Debug.WriteLine("SendFileAsync() - final.");
-            }
-
-            App.Controller.TelemetryClient.TrackEvent("OneDrive picture upload success", properties);
-        }
-
-        internal async Task DebugTextResultAsync(HttpResponseMessage response)
-        {
-            //Debug statements
-            string Text = SerializeHeaders(response);
-            string responseBodyAsText = await response.Content.ReadAsStringAsync().AsTask(cts.Token);
-            cts.Token.ThrowIfCancellationRequested();
-            responseBodyAsText = responseBodyAsText.Replace("<br>", Environment.NewLine); // Insert new lines.
-
-            Debug.WriteLine("--------------------");
-            Debug.WriteLine(Text);
-            Debug.WriteLine(responseBodyAsText);
-        }
-
-        internal string SerializeHeaders(HttpResponseMessage response)
-        {
-            //Get headers for debug statements
-            StringBuilder output = new StringBuilder();
-            output.Append(((int)response.StatusCode) + " " + response.ReasonPhrase + "\r\n");
-
-            SerializeHeaderCollection(response.Headers, output);
-            SerializeHeaderCollection(response.Content.Headers, output);
-            output.Append("\r\n");
-            return output.ToString();
-        }
-
-        internal void SerializeHeaderCollection(IEnumerable<KeyValuePair<string, string>> headers, StringBuilder output)
-        {
-            foreach (var header in headers)
-            {
-                output.Append(header.Key + ": " + header.Value + "\r\n");
-            }
-        }
-
-        private async void refreshTimer_Tick(object sender, object e)
-        {
-            //Refresh OneDrive login tokens every 25 minutes if logged in
-            if (isLoggedIn)
-            {
-                await AuthorizeWithRefreshToken(App.Controller.XmlSettings.OneDriveRefreshToken);
-
-            }
+        private void SaveTokens(object sender, string arg)
+        {
+            App.Controller.XmlSettings.OneDriveAccessToken = oneDriveConnector.accessToken;

```

Comment:
These should probably be saved in PasswordVault instead of plain text XML
http://msdn.microsoft.com/en-us/library/windows/apps/windows.security.credentials.passwordvault.aspx